staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -145,7 +147,10 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 	if s.isResourceRoot {
 		sts = model.WithTypeAndObjectMeta(sts)
 	}
-	activation := NewValidationActivation(obj, sts)
+	var activation interpreter.Activation = NewValidationActivation(ScopedVarName, obj, sts)
```

Comment:
if we're changing `NewValidationActivation` anyway, would it make sense to make it aware of self and oldSelf directly, so this would just be `NewValidationActivation(obj, oldObj, sts)`, and it conditionally sets up oldSelf internally if oldObj is non-nil?

Reply:
also, I'm not sure how expensive the `UnstructuredToVal` call we do on `oldSelf` is when setting up the ValidationActivation, but if none of the rules here have TransitionRule == true, we don't even need oldSelf set up

if the cost of that is non-trivial, would it be worth marking on `Validator` whether we have any transition rules and even need to set up oldSelf?